### PR TITLE
UI overhaul

### LINKED ISSUE #24: publish button for updated queries
- save to library

### LINKED ISSUE #26: re-work query update
- [x] do not re-direct to query index
- [x] run new query
- [x] visualize diff of results:
  - different tweets
  - different slots

### PR DESCRIPTION
**Pull request by mgttlinger**
Optimizing the spheroscope user experience:

- [x] Introduce proper CSS framework
- Make query development more interactive
  - [x] Diffing of intermediate results
  - [x] Timestamps
  - [x] Async result loading
- ~Form validation~
- [x] HTML & JS cleanup 
- [x] Sortable tables

Fixes #24, fixes #26 
Address #24

### CODE REVIEW

**Comment by mgttlinger** [comment]
@ausgerechnet I would need a way to run the intermediate queries (and timestamps in the model) while editing to implement the remaining features. Currently, the `run` function expects the `id` of an existing query. Could this easily be changed to allow to pass the content of an intermediate query?

**Comment by ausgerechnet** [comment]
I'm not 100% sure if I understand correctly. Right now, we have one endpoint (`/queries/<int:id>/run`), which retrieves `<str:cwb_id>` from the app settings and calls `spheroscope.queries.run(id, cwb_id)`. The latter in turn retrieves the query from the database, translates it into a plain dictionary with the following keys:
- cqp
- query > context
- query > s_context
- anchors > corrections
- anchors > slots
- display > p_show
- display > s_show

and finally initializes a `ccc.Corpus` and calls `ccc.cqpy.run_query(corpus, query)`.

So TL;DR: if you're looking for a way to run a query without an ID, make sure the query is a properly formatted `dict`, init a `ccc.Corpus` and directly use `ccc.cqpy.run_query(corpus, query)`.

If you need an API endpoint, just let me know the route and parameters and I'll take care of it.

**Comment by mgttlinger** [comment]
@ausgerechnet I guess I'm somehow messing up the query parameters because running an unchanged query produces longer matches than the stored one. Could you take a look into that?

**Comment by ausgerechnet** [comment]
I'm not sure how to reproduce the bug; I've now unified query execution in commit 63f23e9… let me know if that helps!

Maybe we need unit tests after all …

**Comment by mgttlinger** [comment]
> I'm not sure how to reproduce the bug; I've now unified query execution in commit [63f23e9](https://github.com/ausgerechnet/spheroscope/commit/63f23e90036b82232bc3b9d100f7606014afda96)… let me know if that helps!

Nope still happens on my machine. Steps to reproduce:

1. go to `queries/111/update` 
2. press the "Run" button without doing anything else

This produces results where the first matched tweet `t736396790786850816` matches `&#34` at the end (marked via the dark green colour) while the cached results don't.
On a second look, it seems to always be exactly one token at the end of every extracted slot except 0 leading me to believe that it is related to the corrections not being applied correctly.

**Comment by ausgerechnet** [comment]
I still cannot reproduce the bug.

On my machine, query 111 was "y_will_be_xed_if_z_pattern_2", which doesn't contain tweet t736396790786850816 as a result.

I then realized that you updated patterns.csv in the beginning of June without me re-initializing the instance database – which contains the info from patterns.csv.

Trying to update the database threw a bug since you dropped the "specialises" column; fixed in commit 9569876.

After re-initializing the database, query 111 is "pattern24_adj_affiliaiton_name", which doesn't contain tweet t736396790786850816 as a result either.

As far as I can see, all query results, regardless of whether displayed via `/queries/<id>/run` or `/queries/<id>/update` comprise the exact same spans (matches and slots).
